src/librustc/infer/error_reporting/need_type_info.rs

```diff
@@ -151,12 +151,22 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
         //   |         the type parameter `E` is specified
         // ```
         let (ty_msg, suffix) = match &local_visitor.found_ty {
-            Some(ty) if &ty.to_string() != "_" && name == "_" => {
+            Some(ty) if &ty.to_string() != "_" &&
+                name == "_" &&
+                // FIXME: Remove this check after `impl_trait_in_bindings` is stabilized.
```

Comment:
I believe this is not entirely necessary, given that I added an explicit gate for the feature (which would cause it to be noticed during stabilization), but nonetheless: https://github.com/rust-lang/rust/issues/63527